runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -3974,45 +3974,193 @@ TR::Register *J9::X86::TreeEvaluator::checkcastinstanceofEvaluator(TR::Node *nod
          break;
       }
    auto clazz = TR::TreeEvaluator::getCastClassAddress(node->getChild(1));
-   if (clazz &&
-       !TR::Compiler->cls.isClassArray(cg->comp(), clazz) && // not yet optimized
-       (!cg->comp()->compileRelocatableCode() || cg->comp()->getOption(TR_UseSymbolValidationManager)) &&
-       !cg->comp()->getOption(TR_DisableInlineCheckCast)  &&
-       !cg->comp()->getOption(TR_DisableInlineInstanceOf))
+   if (isCheckCast && !clazz && (!cg->comp()->compileRelocatableCode() || cg->comp()->getOption(TR_UseSymbolValidationManager)))
       {
-      cg->evaluate(node->getChild(0));
-      if (TR::Compiler->cls.isInterfaceClass(cg->comp(), clazz))
-         {
-         generateInlinedCheckCastOrInstanceOfForInterface(node, clazz, cg, isCheckCast);
-         }
-      else
+      auto use64BitClasses = TR::Compiler->target.is64Bit() &&
```

Comment:
The code would look cleaner if these logic is extracted into a separate static function.

Reply:
Refactored into a static function `is64BitClasses`